cfbs/commands.py

```diff
@@ -623,7 +623,10 @@ def _download_dependencies(
         if not os.path.exists(module_dir):
             if url.endswith(SUPPORTED_ARCHIVES):
                 fetch_archive(url, commit)
-            elif "index" in module or ignore_versions:
+            # a couple of cases where there will not be an archive available:
+            # - using an alternate index (index property in module data)
+            # - added by URL instead of name (no version property in module data)
+            elif "index" in module or not "version" in module or ignore_versions:
```

Comment:
The module added by URL should have a `"url"` key.

I think this is the more appropriate way to fix it:

```suggestion
            elif "index" in module or if "url" in module or ignore_versions:
```